labman/gui/testing.py

```diff
@@ -66,5 +66,6 @@ def delete(self, url, data=None, headers=None, doseq=True):
         return self._fetch(url, 'DELETE', headers=headers)
 
     def _fetch(self, url, method, data=None, headers=None):
-        return self.fetch(self.get_url(url), self.stop, method=method,
-                          body=data, headers=headers, timeout=20)
+        self.fetch(self.get_url(url), self.stop, method=method,
+                   body=data, headers=headers)
+        return self.wait(timeout=20)
```

Comment:
...I wasn't able to get this to pass locally as the contents of `self.fetch` needed to be returned to the caller